src/enclave_proc_comm.rs

```diff
@@ -221,14 +215,29 @@ where
             continue;
         }
 
+        let input_stream_raw_fd = events[0].data() as RawFd;
+        let mut input_stream = unsafe { UnixStream::from_raw_fd(input_stream_raw_fd) };
+
         // Handle the reply we received.
-        let mut input_stream = unsafe { UnixStream::from_raw_fd(events[0].data() as RawFd) };
         if let Ok(reply) = read_u64_le(&mut input_stream) {
             if reply == MSG_ENCLAVE_CONFIRM {
                 debug!("Got confirmation from {:?}", input_stream);
                 replies.push(input_stream);
             }
         }
+
+        epoll::epoll_ctl(
+            epoll_fd,
+            EpollOp::EpollCtlDel,
+            input_stream_raw_fd,
+            Option::None,
+        )
+        .map_err(|e| {
+            new_nitro_cli_failure!(
+                &format!("Failed to remove socket from epoll: {:?}", e),
+                NitroCliErrorEnum::EpollError
+            )
+        })?;
```

Comment:
Shouldn't this be in
```rust
           if reply == MSG_ENCLAVE_CONFIRM {
                debug!("Got confirmation from {:?}", input_stream);
                replies.push(input_stream);
            }
```
so that only enclave processes that confirmed the command are removed from the epoll loop? Otherwise, you would remove enclaves that either return an error in read_u64_le or a reply different from MSG_ENCLAVE_CONFIRM

Reply:
No, the mini-protocol expects that the first message should always be "MSG_ENCLAVE_CONFIRM". If something else comes from the connection, we shouldn't wait again on the same socket in case it decides to eventually send a "MSG_ENCLAVE_CONFIRM": if the first message is wrong, then definitely something bad has happened with that enclave, so we should treat it as an error